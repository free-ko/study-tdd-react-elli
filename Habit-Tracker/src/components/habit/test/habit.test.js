import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import renderer from "react-test-renderer";
import Habit from "../habit";

describe("Habit component", () => {
  const habit = { name: "Habit", count: 4 };
  let HabitComponent;
  let onIncrement;
  let onDecrement;
  let onDelete;

  beforeEach(() => {
    onIncrement = jest.fn();
    onDecrement = jest.fn();
    onDelete = jest.fn();

    HabitComponent = (
      <Habit
        habit={habit}
        onIncrement={onIncrement}
        onDecrement={onDecrement}
        onDelete={onDelete}
      />
    );
  });

  it("Renders", () => {
    const component = renderer.create(HabitComponent);

    expect(component.toJSON).toMatchSnapshot();
  });

  describe("Button Click", () => {
    beforeEach(() => {
      render(HabitComponent);
    });

    it('Calls onIncrement when clicking "increment" button', () => {
      const button = screen.getByTitle("increase");
      userEvent.click(button);

      expect(onIncrement).toHaveBeenCalledWith(habit);
    });

    it('Calls onDecrement when clicking "decrement" button', () => {
      const button = screen.getByTitle("decrease");
      userEvent.click(button);

      expect(onDecrement).toHaveBeenCalledWith(habit);
    });

    it('Calls onDelete when clicking "delete" button', () => {
      const button = screen.getByTitle("delete");
      userEvent.click(button);

      expect(onDelete).toHaveBeenCalledWith(habit);
    });
  });
});
