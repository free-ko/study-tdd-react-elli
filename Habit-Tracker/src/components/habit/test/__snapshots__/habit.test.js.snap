// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habit component Renders 1`] = `[Function]`;
